### PR TITLE
Minor Updates: Useless property init; Param docblock; Added Strict Types declaration; Sprintf params, etc.

### PR DESCRIPTION
**Pull request by df2k2**
### Description (*)

**Minor code updates:**

* Added strict types declaration
* Removed useless property initialization where immediately overwritten by declaration in constructor.
* Moved empty array `= []` property initialization to constructor param
* Added missing parameter docblock for constructor
* Updated usage of `sprintf` with mismatched parameter count

### Contribution checklist (*)
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20015) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @df2k2. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by df2k2** [comment]
The Codacy/PR Quality Review fail is for code that was already in place.

`$eavWysiwygDataProcessors` fails on that it exceeds 20 characters.  Not my doing.

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav/CompositeConfigProcessor.php [hunk 1]
```diff
@@ -20,13 +20,15 @@ class CompositeConfigProcessor implements WysiwygConfigDataProcessorInterface
...
 
     /**
      * CompositeConfigProcessor constructor.
-     * @param array $eavWysiwygDataProcessors
+     *
+     * @param Logger $logger
+     * @param array  $eavWysiwygDataProcessors
      */
-    public function __construct(Logger $logger, array $eavWysiwygDataProcessors)
+    public function __construct(Logger $logger, array $eavWysiwygDataProcessors = [])
```
Comment (orlangur) [inline app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav/CompositeConfigProcessor.php]: Such change is not needed, please revert.

File: app/code/Magento/Catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php [hunk 2]
```diff
@@ -3,6 +3,9 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
+declare(strict_types=1);
```
Comment (orlangur) [inline app/code/Magento/Catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php]: There is no need to add strict types in random files, I believe they should be added in bunches of files automatically at some point.

**Review by orlangur — DISMISSED** [review]

File: app/code/Magento/Catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php (hunk 2 above)
Comment (df2k2) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php]: I agree with you.   That was seemingly random on my part.

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav/CompositeConfigProcessor.php (hunk 1 above)
Comment (df2k2) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav/CompositeConfigProcessor.php]: If you insist, I will change that back.  I did update the docblocks for the constructor, so I'll keep that in.

Comment (df2k2) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav/CompositeConfigProcessor.php]: Resolved in commit: 
https://github.com/magento/magento2/pull/20015/commits/b46ea70e214086bcec6f5835781c8802964dc9f5

**Pull request closed without merging**